src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp

```diff
@@ -240,256 +358,202 @@ class boss_gothik : public CreatureScript
-                                    DoTeleportTo(PosGroundDeadSide);
-                                else
-                                    DoTeleportTo(PosGroundLiveSide);
-
-                                me->getThreatManager().resetAggro(NotOnSameSide(me));
-                                if (Unit* target = SelectTarget(SELECT_TARGET_NEAREST, 0))
-                                {
-                                    me->getThreatManager().addThreat(target, 100.0f);
-                                    AttackStart(target);
-                                }
-
-                                events.ScheduleEvent(EVENT_TELEPORT, 20000);
+                                me->StopMoving();
+                                me->SetReactState(REACT_PASSIVE);
+                                me->getThreatManager().resetAllAggro();
+                                DoCastAOE(_lastTeleportDead ? SPELL_TELEPORT_LIVE : SPELL_TELEPORT_DEAD);
+                                _lastTeleportDead = !_lastTeleportDead;
+
+                                events.CancelEvent(EVENT_BOLT);
+                                events.ScheduleEvent(EVENT_TELEPORT, 20 * IN_MILLISECONDS, 0, PHASE_TWO);
+                                events.ScheduleEvent(EVENT_RESUME_ATTACK, 2 * IN_MILLISECONDS, 0, PHASE_TWO);
                             }
                             break;
+
+                        case EVENT_HARVEST:
+                            DoCastAOE(SPELL_HARVEST_SOUL, true); // triggered allows this to go "through" shadow bolt
+                            events.ScheduleEvent(EVENT_HARVEST, 15 * IN_MILLISECONDS, 0, PHASE_TWO);
+                            break;
+                        case EVENT_RESUME_ATTACK:
+                            me->SetReactState(REACT_AGGRESSIVE);
+                            events.ScheduleEvent(EVENT_BOLT, 0, 0, PHASE_TWO);
+                            // return to the start of this method so victim side etc is re-evaluated
+                            return UpdateAI(0u); // tail recursion for efficiency
+                        case EVENT_BOLT:
+                            DoCastVictim(SPELL_SHADOW_BOLT);
+                            events.ScheduleEvent(EVENT_BOLT, 1 * IN_MILLISECONDS, 0, PHASE_TWO);
+                            break;
+                        case EVENT_INTRO_2:
+                            Talk(SAY_INTRO_2);
+                            break;
+                        case EVENT_INTRO_3:
+                            Talk(SAY_INTRO_3);
+                            break;
+                        case EVENT_INTRO_4:
+                            Talk(SAY_INTRO_4);
+                            break;
                     }
                 }
-
-                if (!phaseTwo)
-                    DoMeleeAttackIfReady();
             }
+            
+            private:
+                uint32 _waveCount;
+                bool _gateCanOpen;
+                bool _gateIsOpen;
```

Comment:
V730 It is possible that not all members of a class are initialized inside the constructor. Consider inspecting: _gateIsOpen. boss_gothik.cpp 308